src/InjectHostDevBufferCopies.cpp

```diff
@@ -454,7 +453,56 @@ class InjectBufferCopies : public IRMutator {
             state[buf_name].devices_touched.insert(device_api);
             IRMutator::visit(op);
         } else {
-            IRMutator::visit(op);
+            // PR_TODO(@abadams|@zvookin): At the time of this pass, the IR has
+            // lowered the Call from a Halide one to an Extern one. The
+            // Function is still attached so this seems to work, but this
+            // may be a bit fragile. Is this consistent with the IR design?
+            //
+            // Also a question as to whether this should always use
+            // one path for mutation instead of calling to the parent
+            // class for the regular case. (Always using this path
+            // might catch bugs if the parent class code changes and
+            // this does not track that change.)
+            if (op->call_type == Call::Extern && op->func.defined()) {
+                Function f(op->func);
+
+                internal_assert((f.extern_arguments().size() + f.outputs()) == op->args.size()) <<
+                    "Mismatch between args size and extern_arguments size in call to " << op->name << "\n";
+
+                std::vector<Expr> new_args(op->args.size());
+                bool changed = false;
+                // Mutate the args
+                for (size_t i = 0; i < op->args.size(); i++) {
+                    DeviceAPI old_extern_device_api = extern_device_api;
+                    if (i >= (op->args.size() - f.outputs())) {
```

Comment:
I had asserts for this originally but the outputs() method actually returns an int, not size_t and I was getting warnings. I don't like adding things up to make a thing larger than the index. I'll see if I can make it clearer or at least add a comment. The key thing one has to know is that the outputs are at the end.